### DISCUSSION TITLE
(Monika's body?) Robot running RenPy Scripts

### DISCUSSION THREAD

**Discussion by Delcos** [Monika-After-Story/MonikaModDev#7835] [general - submod / external content]
Hi again,

Recently I've been working with smaller humanoid robots, mainly Aris (seen in the vid), and I'd been fixing the AIML scripts it's running (Artificial Intelligence Markup Language, for it's chatbot) and I didn't have anything to use, so I tried out Monika's replies and conversation topics, and I think it turned out rather well. 

AIML supports tags and other labels so you aren't bound to using specific phrases. An example would be telling her to turn off a light, you could use any structure you want, like: Light off, turn off the light, etc.

We also run the tts voice through pitch modulation and timber flux in software by screaming bee to get an almost GLaDOS like voice. Not sure if that fits the character here but I think it sounds very clear and easy to hear. 

I used a random topic and sentence for this demo, but she's able to respond to almost everything you can ask Monika in MAS through the "talk" button. 
 
Sadly you can't just click and drag the files over, they do have to be remade completely, but the responses and reactions are all the same as MAS. I might continue this project, keeping Aris in character, and if I get enough time maybe write a few responses in MAS about her _"trying out a new body"_  just for fun.  Trying it out with NAO might also be fun but if I do decide to dedicate a robot to this project, then making her a dedicated MAS clone would be overkill, so it will most likely stay on Aris.

_We aren't selling the Renpy scripts or any other content from MAS, so don't worry. This is an internal project and most likely won't ever even be released._ 

**Keep in mind this is a VERY early proof of concept **

https://www.youtube.com/watch?v=p9xt-HDAO_E&feature=youtu.be

**Comment by briaryoung** [comment]
Wow. This is very impressive.

**Comment by bittercaramel** [comment]
ye i like it
pretty neat

**Comment by xXMidnight** [comment]
...what? I don't understand ;-;

**Comment by Delcos** [comment]
> ...what? I don't understand ;-;

This is an older test of running renpy scripts on a robot. "porting" monika to a body. We've now moved to a better body and are testing more in depth responses and reactions. We use the dialog from this mod which is why we post updates about it here.

**Comment by kaio538** [comment]
are you still working on this project? this looks so neat ngl i would love to see more
